atomicfu/build.gradle

```diff
@@ -17,6 +19,12 @@ ext {
     }
 }
 
+// TODO: this block should be removed after the switch to Kotlin language version 1.9
+// (right now it implements a toggle between IR and BOTH JS compiler types for testing of Kotlin dev builds)
+def fromVersion = { String arg -> KotlinVersion.values().find { it.version == arg } }
+def kotlinLanguageVersion = fromVersion(KotlinAggregateBuild.getOverriddenKotlinLanguageVersion(project) ?: "1.4")
+def isJsLegacyCompilerEnabled = kotlinLanguageVersion < KotlinVersion.KOTLIN_1_9
```

Comment:
LanguageVersion has nothing with JS target deprecation. Should be checked using KGP version.

Reply:
I made these changes based on the fact that CLI-level deprecation of JS/Legacy [seems to work based on the language version at the moment](https://github.com/JetBrains/kotlin/blob/552e640517ffa10ab60ba4c6af61e5f18d71cc35/compiler/cli/cli-js/src/org/jetbrains/kotlin/cli/js/K2JSCompiler.java#L189), so we might want to take a closer look at it in the compiler as well then.

Reply:
Hmm 👀 
For now I rolled this back to KGP version check